rosutil/symdebug.rkt

```diff
@@ -0,0 +1,58 @@
+#lang racket
+
+(require (prefix-in r: rosette/safe))
+
+; Utilities for inspecting symbolic expressions in Yosys module state
+
+(provide term-length analyze-terms wide-write)
+
+; helper function that returns flattened length of an arbitrarily nested list
+(define (flattened-length l)
+  (if (not (list? l))
+      1
+      (if (empty? l)
+          0
+          (+ (flattened-length (first l))
+             (flattened-length (rest l))))))
+
+; returns "length" of (the possibly symbolic) t, depending on type of t:
+; vector? returns sum of term-length of each element
+; [symbolic] expression? returns flattened-length of t
+; [symbolic] constant? returns 1
+; other (i.e., concrete value) returns 0
+(define (term-length t)
+  (if (vector? t)
+      (foldl + 0 (map term-length (vector->list t)))
+      (if (r:expression? t)
+          (flattened-length (r:term->datum t))
+          (if (r:term? t)
+              1
+              0))))
+
+; analyze each element in state that is accessed via state-getters
+; state-getters is same construction as the argument of the same name to
+; verify-deterministic-start in rtl/shiva
+;
+; returns number of terms with length >= threshold
+; additionally outputs information about long state components if output? is #t
+(define (analyze-terms state state-getters
+                       #:threshold [threshold 3]
+                       #:output? [output? #f])
```

Comment:
Were you using this in the `#:output? #f` mode? (If it's always useful to see the output, maybe you could make it show unconditionally?)

Reply:
I turned it off when I was using a previous iteration of this function to do some debugging, although I agree that it's not super useful to not have output in this function's current form. 

I think one instance where it may be helpful is if you just want to include it as a watchdog for when expressions start to grow over a certain length (but without spewing more output than a single warning from its consumer, perhaps).